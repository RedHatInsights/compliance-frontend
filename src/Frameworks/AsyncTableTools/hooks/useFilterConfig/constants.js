export const TABLE_STATE_NAMESPACE = 'filters';
