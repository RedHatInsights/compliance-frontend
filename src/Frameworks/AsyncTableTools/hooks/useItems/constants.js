export const TABLE_STATE_NAMESPACE = 'items';
