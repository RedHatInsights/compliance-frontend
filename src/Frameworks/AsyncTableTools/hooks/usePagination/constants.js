export const TABLE_STATE_NAMESPACE = 'pagination';
