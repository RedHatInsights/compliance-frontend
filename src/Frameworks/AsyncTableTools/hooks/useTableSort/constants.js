export const TABLE_STATE_NAMESPACE = 'sort';
