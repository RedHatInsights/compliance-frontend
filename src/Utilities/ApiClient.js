class ApiClient {
    DEFAULT_API_HEADERS = {
        'Content-Type': 'application/json',
        Accept: 'application/json'
    }

    constructor(options = {}) {
        this.apiBase = options.apiBase || '/api';
        this.path = options.path || '';
        this.authenticate = options.authenticate || (() => ({}));
        this.onUnauthorised = options.onUnauthorised;
        this.defaultHeaders = options.headers || this.DEFAULT_API_HEADERS;
    }

    async request(path, apiProps, method, options = {}) {
        return await this.callAuthenticate()
        .then(() => this.fetch(path, apiProps, method))
        .then(this.checkForEmptyResponse)
        .then((response) => this.checkForErrors(response, options))
        .then((response) => {
            const json = (async () => await response.json());
            return json();
        })
        .catch(this.finalCatch);
    }

    finalCatch(promise) {
        if (promise.errors === undefined) {
            return Promise.reject({ title: 'Error parsing' });
        } else {
            return Promise.reject({ ...promise });
        }
    }

    fetch(path, apiProps, method) {
        let params = {
            method: method || 'get',
            headers: this.defaultHeaders,
            credentials: 'include'
        };

        if (apiProps) {
            params.body = JSON.stringify(apiProps);
        }

        return fetch(
            `${ this.apiBase }${ this.path ? this.path : '' }${ path ? path : '' }`, params
        );
    }

    checkForEmptyResponse(response) {
        return response.status === 204 ? { json: () => ({}) } : response;
    }

    checkForErrors(response, options = {}) {
        if (response.status === 404 && options.ignore404) {
            return { json: () => ({}) };
        }

        if (response.status === 401) {
            return this.onUnauthorised();
        }

        const responseCloneJson = response.clone ? response.clone().json() : response;

        if (response.status === 422) {
            return responseCloneJson.then((json) =>
                Promise.reject({ ...json, title: 'Validation error' })
            );
        }

        if (response.status >= 400 && response.status <= 600) {
            return responseCloneJson.then((json) =>
                Promise.reject(json.errors[0])
            );
        }

        return response;
    }

    callAuthenticate() {
        return Promise.resolve(this.authenticate() || true);
    }

    create(path, apiProps) {
        return this.request(path, apiProps, 'post');
    }

    update(path, apiProps) {
        return this.request(path, apiProps, 'put');
    }

    get(path, options = {}) {
        return this.request(path, null, 'get', options);
    }

    destroy(path) {
        return this.request(path, null, 'delete');
    }
}

export default ApiClient;
