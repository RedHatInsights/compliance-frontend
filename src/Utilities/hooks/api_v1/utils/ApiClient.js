class ApiClient {
  DEFAULT_API_HEADERS = {
    'Content-Type': 'application/vnd.api+json',
    Accept: 'application/json',
  };

  constructor(options = {}) {
    this.apiBase = options.apiBase || '/api';
    this.path = options.path || '';
    this.authenticate = options.authenticate || (() => Promise.resolve({}));
    this.onUnauthorised = options.onUnauthorised || (() => Promise.resolve({}));
    this.defaultHeaders = options.headers || this.DEFAULT_API_HEADERS;
  }

  async request(path, apiProps, method, options = {}) {
    return await this.callAuthenticate()
      .then(() => this.fetch(path, apiProps, method, options))
      .then(this.checkForEmptyResponse)
      .then((response) => this.checkForErrors(response, options))
      .then((response) => response.json())
      .catch(this.finalCatch);
  }

  finalCatch(promise) {
    if (promise.errors === undefined) {
      return Promise.reject({ title: 'Error parsing' });
    } else {
      return Promise.reject({ ...promise });
    }
  }

  fetch(path, apiProps, method, options) {
    let params = {
      method: method || 'get',
      headers: this.defaultHeaders,
      credentials: 'include',
    };
    const fullPath = `${this.apiBase}${this.path ? this.path : ''}${
      path ? path : ''
    }`;
    let url = new URL(fullPath, window.location.origin);
    if (options.params) {
      const params = Object.keys(options.params)
        .map(
          (key) =>
            encodeURIComponent(key) +
            '=' +
            encodeURIComponent(options.params[key])
        )
        .join('&');
      url.search = new URLSearchParams(params);
    }

    if (apiProps) {
      params.body = JSON.stringify(apiProps);
    }

    return fetch(url, params);
  }

  checkForEmptyResponse(response) {
    return response.status === 204 ? { json: () => ({}) } : response;
  }

  checkForErrors(response, options = {}) {
    if (response.status === 404 && options.ignore404) {
      return { json: () => ({}) };
    }

    if (response.status === 401) {
      return this.onUnauthorised();
    }

    const responseCloneJson = response.clone
      ? response.clone().json()
      : response;

    if (response.status === 422) {
      return responseCloneJson.then((json) =>
        Promise.reject({ ...json, title: 'Validation error' })
      );
    }

    if (response.status >= 400 && response.status <= 600) {
      return responseCloneJson.then((json) => Promise.reject(json.errors[0]));
    }

    return response;
  }

  callAuthenticate() {
    return Promise.resolve(this.authenticate() || true);
  }

  create(path, apiProps) {
    return this.request(path, apiProps, 'post');
  }

  update(path, apiProps) {
    return this.request(path, apiProps, 'put');
  }

  get(path, options = {}) {
    return this.request(path, null, 'get', options);
  }

  destroy(path) {
    return this.request(path, null, 'delete');
  }
}

export default ApiClient;
