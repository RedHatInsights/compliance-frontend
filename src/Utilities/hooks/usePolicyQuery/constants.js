import gql from 'graphql-tag';

export const POLICY_QUERY = gql`
  query PQ_Profile($policyId: String!) {
    profile(id: $policyId) {
      id
      name
      refId
      external
      description
      totalHostCount
      compliantHostCount
      complianceThreshold
      osMajorVersion
      supportedOsVersions
      lastScanned
      policyType
      policy {
        id
        name
        refId
        profiles {
          id
          name
          refId
          osMinorVersion
          osMajorVersion
          parentProfileId
          benchmark {
            id
            title
            latestSupportedOsMinorVersions
            osMajorVersion
            version
            profiles {
              id
              refId
              ssgVersion
            }
          }
          rules {
            id
            title
            severity
            rationale
            refId
            description
            remediationAvailable
            references
            identifier
            precedence
            values
          }
        }
      }
      businessObjective {
        id
        title
      }
      hosts {
        id
        osMinorVersion
        osMajorVersion
      }
    }
  }
`;

export const POLICY_RULE_TREES_QUERY = gql`
  query PQ_Profile($policyId: String!) {
    profile(id: $policyId) {
      policy {
        profiles {
          id
          benchmark {
            ruleTree
          }
        }
      }
    }
  }
`;

export const POLICY_VALUE_DEFINITONS_QUERY = gql`
  query PQ_Profile($policyId: String!) {
    profile(id: $policyId) {
      policy {
        profiles {
          id
          values
          benchmark {
            valueDefinitions {
              id
              refId
              title
              valueType
              defaultValue
              description
            }
          }
        }
      }
    }
  }
`;
