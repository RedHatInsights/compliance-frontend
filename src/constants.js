import { version } from './../package.json';
import { conditionalFilterType } from '@redhat-cloud-services/frontend-components';

export const COMPLIANCE_API_ROOT = '/api/compliance';
export const COMPLIANCE_UI_ROOT = '/rhel/compliance';
export const COMPLIANCE_WS_ROOT = process.env.NODE_ENV === 'production'
    ? 'wss://localhost:3000/cable'
    : 'ws://localhost:3000/cable';
export const INVENTORY_API_ROOT = '/api/inventory/v1';

export const API_HEADERS = {
    'X-Insights-Compliance': version,
    'Content-Type': 'application/json',
    Accept: 'application/json'
};

export const DEFAULT_SYSTEMS_FILTER_CONFIGURATION = [
    {
        type: conditionalFilterType.text,
        label: 'Name',
        event: 'onSubmit',
        filterString: (value) => (`name ~ ${value}`)
    }
];

export const COMPLIANT_SYSTEMS_FILTER_CONFIGURATION = [
    {
        type: conditionalFilterType.checkbox,
        label: 'Compliant',
        filterString: (value) => (`compliant = ${value}`),
        items: [
            { label: 'Compliant', value: 'true' },
            { label: 'Non-compliant', value: 'false' }
        ]
    },
    {
        type: conditionalFilterType.checkbox,
        label: 'Compliance score',
        filterString: (value) => {
            const scoreRange = value.split('-');
            return `compliance_score >= ${scoreRange[0]} and compliance_score <= ${scoreRange[1]}`;
        },
        items: [
            { label: '90 - 100%', value: '90-100' },
            { label: '70 - 89%', value: '70-89' },
            { label: '50 - 69%', value: '50-69' },
            { label: 'Less than 50%', value: '0-49' }
        ]
    }
];
