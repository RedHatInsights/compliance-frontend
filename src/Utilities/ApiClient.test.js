import ApiClient from './ApiClient';
import { COMPLIANCE_API_ROOT } from '@/constants';

describe('ApiClient', () => {
    it('returns an instance', () => {
        const client = new ApiClient({
            apiBase: COMPLIANCE_API_ROOT,
            authenticate: () => (
                console.log('lol')
            ),
            onUnauthorised: () => {
                console.log('logout')
            }
        });

        expect(client).toMatchSnapshot();
    });
});
