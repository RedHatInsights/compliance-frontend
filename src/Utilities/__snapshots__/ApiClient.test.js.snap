// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiClient returns an instance 1`] = `
ApiClient {
  "apiBase": "/api/compliance",
  "authenticate": [Function],
  "defaultHeaders": Object {
    "Accept": "application/json",
    "Content-Type": "application/json",
  },
}
`;
